pkg/sessions/redis/lock.go

```diff
@@ -0,0 +1,69 @@
+package redis
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/bsm/redislock"
+	"github.com/go-redis/redis/v8"
+	"github.com/oauth2-proxy/oauth2-proxy/v7/pkg/apis/sessions"
+)
+
+var LockPrefix = "lock"
+
+type Lock struct {
+	client redis.Cmdable
+	locker *redislock.Client
+	lock   *redislock.Lock
+	key    string
+}
+
+func NewLock(client redis.Cmdable, key string) sessions.Lock {
+	return &Lock{
+		client: client,
+		locker: redislock.New(client),
+		key:    key,
+	}
+}
+
+func (l *Lock) Obtain(ctx context.Context, expiration time.Duration) error {
+	lock, err := l.locker.Obtain(ctx, l.lockKey(), expiration, nil)
+	if err == redislock.ErrNotObtained {
```

Comment:
Just in case there's any wrapping going on here, is it worth using `errors.Is()`?